crates/interpreter/src/instructions/host.rs

```diff
@@ -521,3 +547,47 @@ pub fn call_inner<SPEC: Spec>(
         }
     }
 }
+
+pub fn call_inner<SPEC: Spec>(
+    interpreter: &mut Interpreter,
+    scheme: CallScheme,
+    host: &mut dyn Host,
+) {
+    match scheme {
+        CallScheme::DelegateCall => check!(interpreter, SPEC::enabled(HOMESTEAD)), // EIP-7: DELEGATECALL
+        CallScheme::StaticCall => check!(interpreter, SPEC::enabled(BYZANTIUM)), // EIP-214: New opcode STATICCALL
+        _ => (),
+    }
+    interpreter.return_data_buffer = Bytes::new();
+
+    let mut out_offset: usize = 0;
+    let mut out_len: usize = 0;
+    let mut call_input: Option<Box<CallInputs>> = None;
+    prepare_call_inputs::<SPEC>(
+        interpreter,
+        scheme,
+        host,
+        &mut out_len,
+        &mut out_offset,
+        &mut call_input,
+    );
+
+    if call_input.is_none() {
+        return;
+    }
+
+    // The Option is not needed form this point
+    let mut call_input = call_input.unwrap();
```

Comment:
Can use `let else` here.
```suggestion
    if call_input.is_none() {
        return;
    }

    // The Option is not needed form this point
    let Some(mut call_input) = call_input else { return };
```